sonar-php-plugin/src/main/java/org/sonar/plugins/php/reports/phpunit/CoverageFileParserForPhpUnit.java

```diff
@@ -66,13 +66,20 @@ private static void parse(List<PackageNode> packages, Consumer<FileNode> consume
 
   private static CoverageNode getCoverage(File report) throws ParseException, IOException {
     SMInputFactory inputFactory = JUnitLogParserForPhpUnit.inputFactory();
+    SMHierarchicCursor rootCursor = null;
     try {
-      SMHierarchicCursor rootCursor = inputFactory.rootElementCursor(report);
-      rootCursor.advance();
-      if (!"coverage".equals(rootCursor.getLocalName())) {
-        throw new XMLStreamException("Report should start with <coverage>");
+      try {
+        rootCursor = inputFactory.rootElementCursor(report);
+        rootCursor.advance();
+        if (!"coverage".equals(rootCursor.getLocalName())) {
+          throw new XMLStreamException("Report should start with <coverage>");
+        }
+        return parseCoverageNode(rootCursor);
+      } finally {
+        if (rootCursor != null) {
+          rootCursor.getStreamReader().closeCompletely();
+        }
       }
```

Comment:
Not sure why the nested `try{}catch`, I would have added the `finally` clause in the parent `try{}catch`.

Reply:
Because `closeCompletely` can throw another `XMLStreamException` on its own, and we are wrapping them into `ParsingException`s here